feat: user journeys

CAMOBAP opened the pull request.

CAMOBAP committed:
feat: user journeys

e271828- commented on HCaptcha/Classes/Journeylitics/FieldKey.swift:
```diff
@@ -0,0 +1,63 @@
+    case x = "x"
+    case y = "y"
+    case index = "idx"
+    case section = "sct"
```
```suggestion
    case section = "sect"
```

mismatch to above ?

CAMOBAP replied:
It's actually short name we can keep any of it

e271828- reviewed (commented):
looks good overall. left minor comments, also:

JLCore.stop() removes observers + clears sinks but does not unswizzle. if we can't safely do this, should at least document that behavior. if multiple HCaptcha instances call start/stop, can cause stopped-after-one-instance-stops issue.

CAMOBAP commented on HCaptcha/Classes/Journeylitics/Hooks/UIGestureRecognizer+JL.swift:
```diff
@@ -0,0 +1,33 @@
+    @objc func jl_handleGesture(_ gesture: UIGestureRecognizer) {
+        let meta = createFieldMap((.gesture, String(describing: type(of: gesture))))
+        JLCore.shared.emit(kind: .gesture, view: gesture.view!.viewTypeName, metadata: meta)
+    }
```
Fixed

CAMOBAP committed:
fix: compilation issues for HCaptcha+Rx__Tests.swift

CAMOBAP committed:
feat: add Carthage target for Jourleylitics

CAMOBAP committed:
chore: update after rebase and call setData properly

CAMOBAP committed:
chore: stabilize JourneyliticsHooks__Tests

CAMOBAP committed:
fix: resolve swift-lint issues, increase pod size

- Fix all SwiftLint violations (line length, alignment, trailing newlines)
- Increase fastlane pod size limit to 200KB
- Refactor HCaptchaWebViewManager to reduce file size (410→331 lines)
- Fix Carthage Xcode to compile on 14.x version

brdlyptrs commented on HCaptcha/Assets/journeys.schema.json:
```diff
@@ -0,0 +1,40 @@
+  "$id": "https://hcaptcha.com/schemas/ios/journeys.schema.json",
+  "title": "HCaptcha Journeys Events",
+  "type": "array",
+  "items": {
```
Is there any value to using single characters or acronyms instead of defining the properties fully?

brdlyptrs commented on HCaptcha/Classes/HCaptchaVerifyParams.swift:
```diff
@@ -69,14 +100,17 @@ public class HCaptchaVerifyParams: NSObject {
             dict["rqdata"] = rqdata
         }
+        if let userJourneys = userJourneys {
+            dict["user_journeys"] = userJourneys
```
Mind if we use the camel case method like you have as the variable already? We do it for every other parameter, although MFA ones seemed to have deviated some how.

CAMOBAP committed:
chore: use camelcase userJourney

- Rename userJourneys to userJourney throughout codebase
- Update dictionary key from user_journeys to userJourney
- Add user journey documentation section to README

CAMOBAP committed:
chore: update key names in journeys.schema.json according feedback

CAMOBAP committed:
chore: update timestamp and support string/map metadata

CAMOBAP committed:
chore: if journey disabled don't assign empty array

CAMOBAP committed:
feat: use begin/end callbacks to emit drag events

CAMOBAP committed:
fix: update RxSwift module for Swift 5.9 support

CAMOBAP committed:
chore: test RxSwift for flatMap inference on private fork

CAMOBAP commented:
Blocked by https://github.com/ReactiveX/RxSwift/pull/2689

CAMOBAP committed:
fix: pod_link expression implicitly coerced from '[Any]?' to 'Any'

e271828- committed:
Fix verify param forwarding on initial load

Keep execute(...) passing the public verify params schema from Swift and
translate those values to the hcaptcha.setData(...) wire format inside the
HTML bridge instead. This preserves deprecated config-level rqdata as a
fallback, lets runtime verifyParams.rqdata override it, forwards MFA values
as mfa_phoneprefix and mfa_phone, and replays the same verify params after
DidLoad so the initial-load path behaves the same as later executions.

Separate reset-on-error from verify-param presence so non-verify-params
validation and passive-key paths do not rely on deprecated helper behavior,
and reject empty verify-param payloads before they reach JS. Update the mock
page and unit coverage to validate the production bridge directly, including
rqdata fallback and override, MFA payload translation, user-journey
injection, and the revised manager/reset semantics.

Tested with bundle exec fastlane test.

e271828- committed:
test cleanup
